### PR TITLE
fix: Respect graceful mode on init (from sdk-common-jvm)

### PR DESCRIPTION
**Pull request by typotter**
## Changes
- Update to latest `sdk-common-jvm`
- Tests for ensuring EppoClient still gives assignment if it fails init.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ test-data:
 	mkdir -p ${tempDir}
 	git clone -b ${branchName} --depth 1 --single-branch ${githubRepoLink} ${gitDataDir}
 	cp -r ${gitDataDir}/ufc ${testDataDir}
-	rm ${testDataDir}/ufc/bandit-tests/*.dynamic-typing.json
+	rm -f ${testDataDir}/ufc/bandit-tests/*.dynamic-typing.json
```
Comment (typotter) [inline Makefile:43]: prevents failure to delete files from bubbling up to test runners

File: src/main/java/cloud/eppo/EppoClient.java [hunk 2]
```diff
@@ -197,8 +197,17 @@ public EppoClient buildAndInit() {
               pollingIntervalMs,
               pollingIntervalMs / DEFAULT_JITTER_INTERVAL_RATIO);
 
-      // Kick off the first fetch
-      fetchConfigurationsTask.run();
+      fetchConfigurationsTask.scheduleNext();
+
+      // Kick off the first fetch, respecting graceful mode.
+      try {
+        instance.loadConfiguration();
```
Comment (typotter) [inline src/main/java/cloud/eppo/EppoClient.java:204]: This actually makes the `buildAndInit` method blocking and guarantees the client is ready to make assignments immediately after calling

Comment (typotter) [reply src/main/java/cloud/eppo/EppoClient.java:204]: Is this OK to do?

Comment (felipecsl) [reply src/main/java/cloud/eppo/EppoClient.java:204]: If I understand correctly, the old behavior here was to kick off a background thread to perform the init and return immediately, while it now blocks the thread and doesn't return until `loadConfiguration` finished. is that correct?
if so, seems like an API breaking change to me

Comment (typotter) [reply src/main/java/cloud/eppo/EppoClient.java]: That's correct.
I have to agree on your assessment of this being a breaking change; while it doesn't break the API signature, it breaks the contract of the API (non blocking, not guaranteed to be ready to make assignments).

I've updated to keep the current behaviour in place: clients is not guaranteed to be ready after init (there's no ability to block-until-ready or get a callback), AND, graceful mode is intrinsically true during Init.
We can revisit these with the next major bump.

**Review by felipecsl — APPROVED** [review]

**Pull request merged**